zhaquirks/xiaomi/aqara/switch_acn047.py

```diff
@@ -0,0 +1,173 @@
+        Pulse = 0x01
+        Dry = 0x03
+
+    attributes = {
+        0x000A: ("switch_type", SwitchType, True),
+        0x0517: ("startup_on_off", StartupOnOff, True),
+        0x0200: ("decoupled_mode", DecoupledMode, True),
+        0x02D0: ("interlock", t.Bool, True),
+        0x0289: ("mode", Mode, True),
+        0x00EB: ("pulse_length", t.uint16_t, True),
+    }
+
+
+class AqaraT2Relay(XiaomiCustomDevice):
+    """Aqara T2 in-wall relay device."""
+
+    signature = {
+        MODELS_INFO: [("Aqara", "lumi.switch.acn047")],
+        ENDPOINTS: {
+            # <SimpleDescriptor endpoint=1 profile=260 device_type=0
+            # device_version=1
+            # input_clusters=[0, 3, 4, 5, 6, 18, 1794, 2820, 64704]
+            # output_clusters=[10, 25]>
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Identify.cluster_id,
+                    Scenes.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    MultistateInput.cluster_id,
+                    Metering.cluster_id,
+                    ElectricalMeasurement.cluster_id,
+                    OppleCluster.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [Time.cluster_id, Ota.cluster_id],
+            },
+            # <SimpleDescriptor endpoint=2 profile=260 device_type=0
+            # device_version=1
+            # input_clusters=[3, 4, 5, 6, 18, 64704]
+            # output_clusters=[]>
+            2: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    Identify.cluster_id,
+                    Scenes.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    MultistateInput.cluster_id,
+                    OppleCluster.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [],
+            },
+            # <SimpleDescriptor endpoint=21 profile=260 device_type=0
+            # device_version=1
+            # input_clusters=[12]
+            # output_clusters=[]>
+            21: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [AnalogInput.cluster_id],
+                OUTPUT_CLUSTERS: [],
+            },
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    BasicCluster,
+                    Identify.cluster_id,
+                    Scenes.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    MultistateInput.cluster_id,
```

Comment:
I think most Aqara devices use the custom `MultistateInputCluster` cluster from `opple_remote`, but I'm not sure we need it.

Reply:
I don't have one of these to test.... I wonder if this is for events if the switch is in decoupled mode? If so we would probably need a custom one?

Reply:
[It](https://github.com/zigpy/zha-device-handlers/blob/a0c41091e857bd28a070b96dc065f2fe29f08539/zhaquirks/xiaomi/aqara/opple_remote.py#L102-L133) disables setting up attribute reporting and sends ZHA events depending on an attribute update.
So yeah, seems to be for decoupled mode.

From a quick look, Z2M seems to do *something* with that cluster: https://github.com/Koenkk/zigbee-herdsman-converters/blob/ab11b8a796bab2620e233e5a7a0b92506c6d6639/src/lib/xiaomi.ts#L1378-L1383

Should be fine to merge this as-is of course. If it's needed in the future, it can be added (or adapted if the T2 relay works differently).

Reply:
done